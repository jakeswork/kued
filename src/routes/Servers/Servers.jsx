import React from 'react';

import { propTypes, defaultProps } from './types';
import MainLayout from '../../layouts/MainLayout';

const Servers = ({ classes }) => (
  <MainLayout>
    Servers
  </MainLayout>
);

Servers.propTypes = propTypes;

Servers.defaultProps = defaultProps;

export default Servers;
