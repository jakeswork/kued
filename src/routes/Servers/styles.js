export default {
  serverInfo: {
    margin: '16px 0',
  },
  grid: {
    display: 'flex',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    margin: '16px 0',
  },
};
