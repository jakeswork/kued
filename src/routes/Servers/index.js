import withStyles from 'react-jss';

import Servers from './Servers';
import styles from './styles';

export default withStyles(styles)(Servers);
