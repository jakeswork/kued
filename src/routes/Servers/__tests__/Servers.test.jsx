import React from 'react';
import { shallow } from 'enzyme';

import Servers from '../Servers';

const defaultProps = {
  classes: {},
};

describe('The Servers instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<Servers {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when typing in the input to search', () => {
    it('should update the state and limit the cards displayed', () => {
      const wrapper = shallow(<Servers {...defaultProps} />);
      const input = wrapper.find({ 'data-test-id': 'searchInput' });

      input.simulate('change', { target: { value: 'Test' } });

      expect(wrapper.state('searchInput')).toEqual('test');
    });
  });
});
