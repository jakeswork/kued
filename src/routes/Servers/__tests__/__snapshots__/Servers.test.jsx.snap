// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Servers instance When called with default props should render without throwing errors 1`] = `
<Jss(MainLayout)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={true}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    Servers
  </Jss(Text)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={true}
    link={null}
    style={Object {}}
  >
    Kued is bespoke software. Below you can find a list of currently supported servers, as well as the status of other servers.
  </Jss(Text)>
  <Jss(Input)
    className=""
    data-test-id="searchInput"
    onChange={[Function]}
    placeholder="Search servers"
  />
  <div>
    <Jss(Card)
      className={null}
      key="Warmane"
      style={
        Object {
          "width": 275,
        }
      }
    >
      <div>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={true}
          h5={false}
          link={null}
          style={Object {}}
        >
          Warmane
        </Jss(Text)>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link="https://www.warmane.com/"
          style={Object {}}
        >
          https://www.warmane.com/
        </Jss(Text)>
      </div>
      <Jss(Chip)
        amber={false}
        green={true}
        key="TBC"
        red={false}
        tooltip="Supported"
      >
        TBC
      </Jss(Chip)>
      <Jss(Chip)
        amber={false}
        green={true}
        key="WoTLK"
        red={false}
        tooltip="Supported"
      >
        WoTLK
      </Jss(Chip)>
      <Jss(Chip)
        amber={false}
        green={false}
        key="MoP"
        red={false}
        tooltip="Development"
      >
        MoP
      </Jss(Chip)>
    </Jss(Card)>
    <Jss(Card)
      className={null}
      key="WoW Circle"
      style={
        Object {
          "width": 275,
        }
      }
    >
      <div>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={true}
          h5={false}
          link={null}
          style={Object {}}
        >
          WoW Circle
        </Jss(Text)>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link="https://www.wowcircle.com/en/"
          style={Object {}}
        >
          https://www.wowcircle.com/en/
        </Jss(Text)>
      </div>
      <Jss(Chip)
        amber={true}
        green={false}
        key="WotLK"
        red={false}
        tooltip="Considered"
      >
        WotLK
      </Jss(Chip)>
    </Jss(Card)>
    <Jss(Card)
      className={null}
      key="TwinStar"
      style={
        Object {
          "width": 275,
        }
      }
    >
      <div>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={true}
          h5={false}
          link={null}
          style={Object {}}
        >
          TwinStar
        </Jss(Text)>
        <Jss(Text)
          bold={false}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link="https://www.twinstar.cz/"
          style={Object {}}
        >
          https://www.twinstar.cz/
        </Jss(Text)>
      </div>
      <Jss(Chip)
        amber={true}
        green={false}
        key="WoTLK"
        red={false}
        tooltip="Considered"
      >
        WoTLK
      </Jss(Chip)>
      <Jss(Chip)
        amber={true}
        green={false}
        key="Cata"
        red={false}
        tooltip="Considered"
      >
        Cata
      </Jss(Chip)>
    </Jss(Card)>
  </div>
</Jss(MainLayout)>
`;
